src/main/java/seedu/address/model/commission/Commission.java

```diff
@@ -0,0 +1,139 @@
+package seedu.address.model.commission;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.customer.Customer;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a Commission in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Commission {
+
+    // Identity fields
+    private final Title title;
+    private final Description description;
+    private final Fee fee;
+    private final Deadline deadline;
+    private final CompletionStatus completionStatus;
+    private final Set<Tag> tags = new HashSet<>();
+    private Customer customer;
```

Comment:
Could we make this immutable instead?

Reply:
I am not sure how to make this immutable...

The main issue I am having is in `JsonAdaptedCustomer.java` where if I create the Commission first, I cannot set it to the customer created after. If I create the customer first, that would mean the commission set would have to be mutable.

Reply:
I see, in that case, we can just leave it. Thanks!